megamek/src/megamek/common/pathfinder/CachedEntityState.java

```diff
@@ -132,4 +135,33 @@ public boolean isAmphibious() {
                 hasWorkingMisc(MiscType.F_AMPHIBIOUS) ||
                 hasWorkingMisc(MiscType.F_LIMITED_AMPHIBIOUS);
     }
+    
+    /**
+     * Convenience property to determine how many armor-breached legs this entity has.
+     * By default, this is 0 unless the unit is a mech.
+     */
+    public int getNumBreachedLegs() {
+        if (numBreachedLegs == null) {
+            if (backingEntity instanceof QuadMech) {
+                numBreachedLegs = 
+                        ((backingEntity.getArmor(QuadMech.LOC_LLEG) > 0) ? 0 : 1) +
+                        ((backingEntity.getArmor(QuadMech.LOC_LARM) > 0) ? 0 : 1) +
+                        ((backingEntity.getArmor(QuadMech.LOC_RLEG) > 0) ? 0 : 1) +
+                        ((backingEntity.getArmor(QuadMech.LOC_RARM) > 0) ? 0 : 1);
+            } else if (backingEntity instanceof TripodMech) {
+                numBreachedLegs = 
+                        ((backingEntity.getArmor(TripodMech.LOC_LLEG) > 0) ? 0 : 1) +
+                        ((backingEntity.getArmor(TripodMech.LOC_CLEG) > 0) ? 0 : 1) +
+                        ((backingEntity.getArmor(TripodMech.LOC_RLEG) > 0) ? 0 : 1);
+            } else if (backingEntity instanceof Mech) {
+                numBreachedLegs = 
+                        ((backingEntity.getArmor(TripodMech.LOC_LLEG) > 0) ? 0 : 1) +
+                        ((backingEntity.getArmor(TripodMech.LOC_RLEG) > 0) ? 0 : 1);
```

Comment:
For what it's worth, lines 158 and 159 could refer to Mech.LOC_ instead of TripodMech.LOC_